### PR TITLE
Issues/83 folders ui

### PR DESCRIPTION
**Pull request by aerych**
Refs #83 

This PR updates the UI of the Story Folder and Story Assets screens as well as enables state restoration when resuming the app.  Note that the ability to rename a story was removed with this PR. There will be a couple of follow up PRs ~to adjust some cell styling~ and to introduce a create/edit screen when making a new folder.  (Edit: Actually the cell styling change was very small so I went ahead and included it.)

To test: 

- [x] Review the Story Folder list.  Confirm that it looks good, no missing icons, etc. 
- [x] Test the sort control. Try sorting by name ascending and descending.
- [x] Confirm that the last selected story is remembered between app launches.
- [x] Confirm that the last selected story is shown in blue text (but not other stories).
- [x] Review the Story Asset list.  Confirm that it looks good, no missing icons, etc. 
- [x] Confirm sorting still works.  Note that the test behavior of tapping rows to move between sorted and unsorted lists is still in place.  New text notes may be added by tapping the text note button on the Assets screen. 
- [x] Confirm the title of the assets screen is the name of the story you chose.
- [x] While viewing the assets screen, background the app to trigger state restoration (remember to wait a second or two). Stop debugging, the relaunch the app.  Confirm you are returned to the assets screen for the story you had selected.
- [x] Confirm darkmode looks good on both screens. 

@jleandroperez how about another? 

### Examples:
Simulator Screen Shot - iPhone SE (2nd generation) - 2020-07-28 at 13 46 46
Simulator Screen Shot - iPhone SE (2nd generation) - 2020-07-28 at 12 38 53
Simulator Screen Shot - iPhone SE (2nd generation) - 2020-07-28 at 13 46 54

### CODE REVIEW

File: Newspack/Newspack/Controllers/AssetsViewController.swift [hunk 1]
```diff
@@ -2,9 +2,22 @@ import UIKit
...
+    @IBOutlet var syncButton: UIBarButtonItem!
+    @IBOutlet var editButton: UIBarButtonItem!
+    @IBOutlet var tableView: UITableView!
+
+    @IBOutlet var textNoteButton: UIBarButtonItem!
+    @IBOutlet var photoButton: UIBarButtonItem!
+    @IBOutlet var videoButton: UIBarButtonItem!
+    @IBOutlet var audioNoteButton: UIBarButtonItem!
 
     var dataSource: AssetDataSource!
```
Comment (jleandroperez) [inline Newspack/Newspack/Controllers/AssetsViewController.swift]: (Not added here!!) but should this one be private perhaps?

File: Newspack/Newspack/Controllers/FoldersViewController.swift [hunk 2]
```diff
@@ -1,68 +1,143 @@
...
-    var dataSource: FolderDataSource!
+    @IBOutlet var textNoteButton: UIBarButtonItem!
+    @IBOutlet var photoButton: UIBarButtonItem!
+    @IBOutlet var videoButton: UIBarButtonItem!
+    @IBOutlet var audioNoteButton: UIBarButtonItem!
 
-    required init?(coder aDecoder: NSCoder) {
-        super.init(coder: aDecoder)
-    }
+    var dataSource: FolderDataSource!
```
Comment (jleandroperez) [inline Newspack/Newspack/Controllers/FoldersViewController.swift]: (Same note as above!!!) maybe private?

File: Newspack/Newspack/Controllers/MainNavController.swift [hunk 3]
```diff
@@ -31,23 +61,46 @@ final class MainNavController: UINavigationController {
...
     }
+
+}
+
+// MARK: - State Restoration
+
+extension MainNavController: UIViewControllerRestoration {
+    static let sidebarControllerIdentifier = String(describing: type(of: SidebarContainerViewController.self))
+    static let storyNavIdentifier = String(describing: type(of: StoryNavigationController.self))
+    static let menuControllerIdentifier = String(describing: type(of: MenuViewController.self))
```
Comment (jleandroperez) [inline Newspack/Newspack/Controllers/MainNavController.swift]: [Maaaaybe this is useful](https://github.com/Automattic/simplenote-ios/blob/develop/Simplenote/NSObject+Helpers.swift#L10) (OR not!!!)

File: Newspack/Newspack/Controllers/SidebarContainerViewController.swift [hunk 4]
```diff
@@ -193,6 +198,12 @@ class SidebarContainerViewController: UIViewController {
     func handleToggleSidebarNotification(notification: Notification) {
         toggleSidebar()
     }
+
+    override func encodeRestorableState(with coder: NSCoder) {
+        super.encodeRestorableState(with: coder)
+        coder.encode(mainViewController, forKey: Constants.mainControllerEncodeKey)
+        coder.encode(sidebarViewController, forKey: Constants.sideControllerEncodeKey)
+    }
```
Comment (jleandroperez) [inline Newspack/Newspack/Controllers/SidebarContainerViewController.swift:206]: Neat!!!

**Review by jleandroperez — CHANGES_REQUESTED** [review]
### Issue A: TabBar AWOL
I've spotted a weird behavior related to state restoration:

1. Open any story
2. Background the app
3. Wait for 2 seconds
3. Stop Xcode
4. Relaunch the app

As a result, super shortly after the TabBar disappears.
I'm having troubles uploading the video here, I'll DM it over slack!

### Note B: TabBar / Text Note Button
Probably irrelevant ATM, but I just wanted to leave a note on UX: using the bottom left TabBar button to insert a **new Text Note** was a bit counter intuitive. It was literally the last button I've pressed to attempt to add a new Text Note.

Just wanted to pass this note along, **no need to actually change anything here!!!**

### Note C: Contrast
Dark Mode, Stories list. UITableView's highlight color **feels** a bit off.

Again, just a note, no actual need to do anything here.




Back to you sir !!

File: Newspack/Newspack/Controllers/MainNavController.swift (hunk 3 above)
Comment (aerych) [reply Newspack/Newspack/Controllers/MainNavController.swift]: Yeah. Let's do that. Good suggestion

**Comment by aerych** [comment]
> Note A

Well spotted.  Will look into it!

> Note B

Totally placeholder behavior :) 

> Note C

Good catch!

File: Newspack/Newspack/Controllers/MainNavController.swift (hunk 3 above)
Comment (aerych) [reply Newspack/Newspack/Controllers/MainNavController.swift]: Okay... what's funny... is I already had added that extension... and just forgot! 🤦

**Comment by aerych** [comment]
Thank you @jleandroperez!  Ready for another look sir!

**Review by jleandroperez — APPROVED** [review]
✅  State restoration works super
✅  Re-tested all the things, looking sharp!!

Thank you for fixing the cell highlight color!! nice work Eric!!

:shipit:

**Comment by aerych** [comment]
Thank you @jleandroperez !!

**Pull request merged**